src/plugins/data_view_management/public/management_app/data_view_management_service.ts

```diff
@@ -0,0 +1,293 @@
+    if (dataView) {
+      const defaultIndex = await this.services.uiSettings.get('defaultIndex');
+      const tags = getTags(
+        dataView,
+        dataView.id === defaultIndex,
+        this.services.dataViews.getRollupsEnabled()
+      );
+
+      return tags;
+    }
+    return [];
+  };
+
+  async updateScriptedFields() {
+    const dataView = this.state$.getValue().dataView;
+    if (dataView) {
+      const scriptedFieldRecords = dataView.getScriptedFields();
+      const scriptedFields = Object.values(scriptedFieldRecords);
+
+      const scriptedFieldLangs = Array.from(
+        scriptedFields.reduce((acc: Set<string>, field) => {
+          if (field.lang) {
+            acc.add(field.lang);
+          }
+          return acc;
+        }, new Set<string>())
+      );
+
+      this.updateState({
+        scriptedFields,
+        scriptedFieldLangs: convertToEuiFilterOptions(scriptedFieldLangs),
+      });
+    }
+  }
+
+  async setDataView(dataView: DataView) {
+    this.updateState({ isRefreshing: true });
+
+    const fieldRecords = dataView.fields
+      .filter((field) => !field.scripted)
+      .reduce((acc, field) => {
+        acc[field.name] = field;
+        return acc;
+      }, {} as Record<string, DataViewField>);
+
+    const fields = Object.values(fieldRecords);
+    const indexedFieldTypes: string[] = [];
+    fields.forEach((field) => {
+      // for conflicted fields, add conflict as a type
+      if (field.type === 'conflict') {
+        indexedFieldTypes.push('conflict');
+      }
+      if (field.esTypes) {
+        // add all types, may be multiple
+        field.esTypes.forEach((item) => indexedFieldTypes.push(item));
+      }
+    });
+
+    const allowedAsString = (await this.allowedTypes).map((item) => item.name);
+
+    this.services.savedObjectsManagement
+      .getRelationships(DATA_VIEW_SAVED_OBJECT_TYPE, dataView.id!, allowedAsString)
+      .then((resp) => {
+        this.updateState({
+          relationships: resp.relations.map((r) => ({ ...r, title: r.meta.title! })),
+        });
+      });
+
+    this.updateState({
+      dataView,
+      fields,
+      indexedFieldTypes,
+      fieldConflictCount: fields.filter((field) => field.type === 'conflict').length,
+      tags: await this.getTags(dataView),
+      isRefreshing: false,
+      conflictFieldsUrl: this.getKbnUrl(dataView.id!),
+      scriptedFields: dataView.getScriptedFields(),
+    });
+
+    this.updateScriptedFields();
+  }
+
+  async refreshFields() {
+    const dataView = this.state$.getValue().dataView;
+    if (dataView) {
+      await this.services.dataViews.refreshFields(dataView, undefined, true);
+      return this.setDataView(dataView);
+    }
+  }
+
+  async setDefaultDataView() {
+    const dataView = this.internalState$.getValue().dataView;
+    if (!dataView) {
+      return;
+    }
+    await this.services.uiSettings.set('defaultIndex', dataView.id);
+
+    this.updateState({ tags: await this.getTags(dataView) });
```

Comment:
Should we also update `defaultIndex` in the state or can we remove it from the state definition?

Reply:
removed

Reply:
Nope, its used, restored and fixed.